File: Database/teams.js
const { isNil } = require("lodash");

const apicomm = require("./adapters/apihandler");
const dbhandler = require("./adapters/dbhandler.js");
const { fetchTeams } = require("./functions");

async function run() {
  const db = new dbhandler.Database();
  await db.init();

  const teams = await fetchTeams();
  try {
    const collection = db.getCollection("teams");
    for (let team of teams) {
      const location = await apicomm.mapboxApiRequest(
        isNil(team.venue)
          ? team.locationName
          : `${team.venue.name} ${team.venue.city}`
      );

      const options = { upsert: true };
      const filter = { id: team.id };

      const updateDoc = {
        $set: {
          name: team.name,
          active: team.active,
          abbreviation: team.abbreviation,
          locationName: team.locationName,
          description: (await apicomm.wikiApiRequest(team.name)).extract,
          firstYearOfPlay: parseInt(team.firstYearOfPlay),
          franchiseId: team.franchiseId,
          division: team.division,
          conference: team.conference,
          venue: team.venue ? {...team.venue, description: (
            await apicomm.wikiApiAdvancedRequest(team.venue.name, team.venue.city)
          ).extract }: undefined,
          location: location.features.map((el) => {
            return {
              text: el.text,
              placeName: el.place_name,
              center: el.center,
            };
          })[0],
          lastUpdate: new Date(),
        },
      };
      try {
        const queryResult = await collection.updateOne(
          filter,
          updateDoc,
          options
        );
        console.log(
          `${queryResult.matchedCount} document(s) matched the filter, updated ${queryResult.modifiedCount} document(s): ${team.name}`
        );
      } catch (ex) {
        console.log(ex);
      }
    }

    const teamPlayerCollection = db.getCollection("players");

    for (let team of teams) {
      if (team.roster == undefined) {
        continue;
      }
      const roster = team.roster.roster;
      for (let player of roster) {
        const options = { upsert: true };
        const filter = { id: player.person.id };

        const updateDoc = {
          $set: {
            fullName: player.person.fullName,
            jerseyNumber: parseInt(player.jerseyNumber),
            positon: player.position,
            active: true,
          },
        };
        try {
          const queryResult = await teamPlayerCollection.updateOne(
            filter,
            updateDoc,
            options
          );
          console.log(
            `${queryResult.matchedCount} document(s) matched the filter, updated ${queryResult.modifiedCount} document(s): ${player.person.fullName}`
          );
        } catch (ex) {
          console.log(ex);
        }
      }
    }
  } finally {
    await db.closeClient();
  }
}

run().catch(console.dir);

File: Database/teams.py
from datetime import datetime
from apihandler import wiki_api_advanced_request, wiki_api_request
from apihandler import nhl_api_request, mapbox_api_request
from dbhandler import get_collection

def map_location(element):
  return {
    "text": element["text"],
    "placeName": element["place_name"],
    "center": element["center"]
  }

collection = get_collection("teams")
player_collection = get_collection("players")

# fetching all teams
teams = []
for i in range(1,57):
  team = nhl_api_request("/api/v1/teams/{0}".format(i), {"expand": "team.roster"})
  teams.append(team["teams"][0])


for team in teams:
  # get location
  if "venue" in team:
    query = "{0} {1}".format(team["venue"]["name"], team["venue"]["city"])
  else:
    query = team["locationName"]
  location = list(map(map_location, mapbox_api_request(query)["features"]))[0]

  description = wiki_api_request(team["name"]).summary

  if "venue" in team:
    venue = team["venue"]
    venue["description"] = wiki_api_advanced_request(venue["name"], venue["city"]).summary

  
  filter = { "id": team["id"] }
  update = {
    "$set": {
          "name": team["name"],
          "active": team["active"],
          "abbreviation": team["abbreviation"],
          "locationName": team["locationName"],
          "description": description,
          "firstYearOfPlay": int(team["firstYearOfPlay"]) if "firstYearOfPlay" in team else None,
          "franchiseId": team["franchiseId"],
          "division": team["division"],
          "conference": team["conference"],
          "venue": venue,
          "location": location,
          "lastUpdate": datetime.now()
          }
  }
  result = collection.update_one(filter, update, True)
  print("{0} teams found, {1} inserted or updated. team name: {2} ".format(result.matched_count, result.modified_count, team["name"]))
  
  if "roster" in team:
    roster = team["roster"]["roster"]
    for player in roster:
      playerFilter = { "id": player["person"]["id"]}

      updatePlayer = {
          "$set": {
            "fullName": player["person"]["fullName"],
            "jerseyNumber": int(player["jerseyNumber"]) if "jerseyNumber" in player else None,
            "positon": player["position"],
            "active": True,
          },
        };
      
      player_result = player_collection.update_one(playerFilter, updatePlayer, True)
      print("team - player: {0} matched, {1} modified: {2}".format(player_result.matched_count, player_result.modified_count, player["person"]["fullName"]))